demo/src/actions/app-actions.js

```diff
@@ -28,10 +28,25 @@ export const loadContent = filename => {
 
 const loadDataStart = owner => ({type: 'LOAD_DATA_START', owner});
 
-const loadDataSuccess = (owner, data, meta) => ({
-  type: 'LOAD_DATA_SUCCESS',
-  payload: {owner, data, meta}
-});
+const loadDataSuccess = (context, index, data, meta) => {
+
+  if (context.isArray) {
```

Comment:
Apparently this function can handle a couple of different types of data, and expects them to have a certain structure ? Worth documenting?

Also it seems designed to gradually add things to the context parameter (instead of adding them to some state variable). This seems to be outside of the normal redux model and worth commenting).